### PR TITLE
fix: #2138 linter issues with fail function

### PR DESCRIPTION
**Pull request by JuanJo4**
## What does this PR do and why?

It solves #2138, this has happened before (#2137 & #1989) and was solved only for those specific cases, this PR aims to fix once and for all.

This is what I did:
- removed `fail` function and created a new `MstError` class that extends from `Error`.
- adjusted the calls of the `fail` function accordingly.
- removed unused imports

## Steps to validate locally

I've added some basic unit tests, not sure if that's enough.

### CODE REVIEW

**Comment by coolsoftwaretyler** [comment]
Thanks @JuanJo4! I'll take a look today or tomorrow.

File: __tests__/uitls.test.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { MstError } from "../src/utils"
```
Comment (coolsoftwaretyler) [inline __tests__/uitls.test.ts:1]: **issue:** @JuanJo4 - would you mind renaming this file to `__tests__/utils.test.ts`?  I think the file name is a typo currently.

File: __tests__/uitls.test.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+describe("MstError custom error class", () => {
```
Comment (coolsoftwaretyler) [inline __tests__/uitls.test.ts:3]: **praise:** @JuanJo4 - thank you for these test! Make sense to me!

**Review by coolsoftwaretyler — CHANGES_REQUESTED** [review]
Hey @JuanJo4 - I love it. I came around to really liking the new class rather than the function-based approach. Initially, I thought `fail` was `throw`ing, but I had misremenbered. Since we were throwing whatever gets returned from `fail`, I think a new class that extends `Error` is perfect.

I found a typo in one of your file names. Would you mind fixing that? Once you do, we can merge this.

Thanks for doing this! Really appreciate it.

File: __tests__/uitls.test.ts (hunk 1 above)
Comment (JuanJo4) [reply __tests__/uitls.test.ts]: oh! my bad... fixed.

**Comment by coolsoftwaretyler** [comment]
Perfect! Merging.

I'll get a new 6.0.0 release out soon. We're in a preview mode for the next major and this should land there once we ship it.

Thanks for the help, and please let us know if there's anything else you'd like to work on.

**Pull request merged**